docs/html/user_guide.rst

```diff
@@ -855,6 +855,26 @@ We are using `freeze`_ here which outputs installed packages in requirements for
 
   reqs = subprocess.check_output([sys.executable, '-m', 'pip', 'freeze'])
 
+Since pip's progress bar gets hidden when running in a subprocess, you can use
+the ``--progress-bar=json`` option for easily parsable progress information::
+
+  subprocess.check_call([sys.executable, '-m', 'pip', 'install', 'my_package', '--progress-bar=json'])
```

Comment:
This example isn't realistic, as check_call doesn't capture output, it will just dump the JSON onto the caller's `stdout`. I understand that a full example will be complex, but conversely, showing an example that doesn't demonstrate why you would use the feature isn't helpful.

Better would be a discussion that starts something like "If you want to display pip's output in real time..." and explains such issues as needing to ensure the subprocess doesn't buffer its output (the `-u` flag), multiplexing stdout and stderr without getting deadlocks, etc.

Or you could turn the explanation around, and find somewhere to discuss how processing pip's output as it's produced is made more complicated by the way the standard progress bar is written, and use that as a prompt to explain how `--progress-bar=json` provides an alternative format which is easier to process. The problem with that approach is, of course, that we don't currently document processing pip's output in real time...

Reply:
Updated with a more realistic code example based on an actual implementation I did.